src/data/glossary.yml

```diff
@@ -1420,6 +1420,43 @@
     - "pub"
     - "packages"
 
+- term: "Tree shaking"
+  short_description: |-
+    A compiler optimization that removes unused code.
+  long_description: |-
+    **Tree shaking** is a compiler optimization that analyzes which parts of
+    the code are actually used by an application and removes everything else
+    from the final output.
+
+    This makes the compiled program smaller and faster to load, since dead code
+    that is never referenced doesn't end up in the binary or bundle.
+
+    Tree shaking is especially important for web and mobile apps, where code size
+    directly impacts download time and runtime performance.
+
+    Example:
+    If an imported library defines ten classes but your app only uses two,
+    tree shaking ensures the other eight classes are excluded from the compiled output.
```

Comment:
I love that you included this example!

I think it's a bit hard to understand how (and when) classes are "included" in compiled output though, so perhaps it'd be better to replace the mention with functions? Developers often have some familiarity with that from imperative logic.

```
    For example, if you import a library that defines ten top-level functions,
    but your app only calls two, tree shaking ensures
    the other eight functions are excluded from the compiled output.
```